aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,312 @@
+  :lines: 8-35
+  :dedent: 4
+
+.. note:: If you just need to log how long an action takes to execute, rather than displaying it on the page, this is already done by the framework.
+
+.. _exception-filters:
+
+Exception Filters
+-----------------
+*Exception Filters* implement either the ``IExceptionFilter`` or ``IAsyncExceptionFilter`` interface.
+
+Exception filters handle unhandled exceptions. They are only called when an exception occurs later in the pipeline. They can provide a single location to implement common error handling policies within an app. The framework provides an abstract `ExceptionFilterAttribute <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Filters/ExceptionFilterAttribute/index.html>`_ that you should be able to subclass for your needs. Exception filters are good for trapping exceptions that occur within MVC actions, but they're not as flexible as error handling middleware. Prefer middleware for the general case, and use filters only where the middleware doesn't offer the functionality your app requires.
+
+Exception filters do not have two events (for before and after) - they only implement ``OnException`` (or ``OnExceptionAsync``). The ``ExceptionContext`` provided in the ``OnException`` parameter includes the ``Exception`` that occurred. If you set ``context.Exception`` to null, the effect is that you've handled the exception, so the request will proceed as if it hadn't occurred (most likely returning a 200 OK status). The following filter logs exceptions and marks them as handled:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/LoggingExceptionFilterAttribute.cs
+  :language: c#
+  :emphasize-lines: 7,15,18
+
+.. _result-filters:
+
+Result Filters
+--------------
+*Result Filters* implement either the ``IResultFilter`` or ``IAsyncResultFilter`` interface and their execution surrounds the execution of action results. Result filters are only executed for successful results - when the action or action filters produce an action result. **Result filters are not executed when exception filters handle an exception.**
+
+Result filters are ideal for any logic that needs to directly surround view execution or formatter execution. Result filters can replace or modify the action result that's responsible for producing the response.
+
+The ``OnResultExecuting`` method runs before the action result is executed, so it can manipulate the action result through ``ResultExecutingContext.Result``. An ``OnResultExecuting`` method can short-circuit execution of the action result and subsequent result filters by setting ``ResultExecutingContext.Cancel`` to true. If short-circuited, MVC will not modify the response; take care to write to the response object directly when short-circuiting. Throwing an exception in an ``OnResultExecuting`` method will also prevent execution of the action result and subsequent filters, but will be treated as a failure instead of a successful result.
```

Comment:
> If short-circuited, MVC will not modify the response; take care to write to the response object directly when short-circuiting.

If short-circuited, MVC will not modify the response; you should generally write to the response object directly when short-circuiting. (otherwise you'll probably get an empty 200 ???)